test/framework/docker_logcollector.go

```diff
@@ -54,6 +56,35 @@ func (k DockerLogCollector) CollectMachineLog(ctx context.Context, managementClu
 			return execOnContainer(containerName, f, command, args...)
 		}
 	}
+	copyDirFn := func(containerDir, dirName string) func() error {
+		return func() error {
+			f, err := ioutil.TempFile("", containerName)
+			if err != nil {
+				return err
+			}
+
+			tempfileName := f.Name()
+			outputDir := filepath.Join(outputPath, dirName)
+
+			defer os.Remove(tempfileName)
+
+			err = execOnContainer(
+				containerName,
+				f,
+				"tar", "--hard-dereference", "-C", containerDir, "-chf", "-", ".",
```

Comment:
nit:
Just for verbosity and for those that may not know the flags of tar, suggest expanding these to their full flag names. 🙂 

```suggestion
				"tar", "--hard-dereference", "--dereference", "--directory", containerDir,  "--create", "--file", "-", ".",
```

Reply:
Done, expanded the flags on both commands.